PurchasesTests/Purchasing/PurchasesTests.swift

```diff
@@ -2314,6 +2314,41 @@ class PurchasesTests: XCTestCase {
         transaction.setValue(SKPaymentTransactionState.purchased.rawValue, forKey: "transactionState")
         self.storeKitWrapper.delegate?.storeKitWrapper(self.storeKitWrapper, updatedTransaction: transaction)
     }
+    
+    func testNoCrashIfPaymentDoesNotHaveProductIdenfier() {
+        setupPurchases()
+
+        let transaction = MockTransaction()
+        transaction.mockPayment = SKPayment()
+
+        transaction.setValue(SKPaymentTransactionState.purchasing.rawValue, forKey: "transactionState")
+        self.storeKitWrapper.delegate?.storeKitWrapper(self.storeKitWrapper, updatedTransaction: transaction)
+        
+        transaction.setValue(SKPaymentTransactionState.purchased.rawValue, forKey: "transactionState")
+        self.storeKitWrapper.delegate?.storeKitWrapper(self.storeKitWrapper, updatedTransaction: transaction)
+    }
+    
+    func testNilProductIdentifierIfPaymentIsMissing() {
+        let transaction = SKPaymentTransaction()
+        expect(transaction.rc_productIdentifier).to(beNil())
+    }
+    
+    func testNilProductIdentifierIfPaymentDoesNotHaveProductIdenfier() {
+        let transaction = MockTransaction()
+        transaction.mockPayment = SKPayment()
+        
+        expect(transaction.rc_productIdentifier).to(beNil())
+    }
+    
+    func testProductIdentifierFromAnyTransaction() {
+        let expectedProductIdentifier = "com.product.id1"
+        let product = MockSKProduct(mockProductIdentifier: expectedProductIdentifier)
+        let payment = SKPayment(product: product)
+        let transaction = MockTransaction()
+        transaction.mockPayment = payment
+        
+        expect(transaction.rc_productIdentifier).to(equal(expectedProductIdentifier))
+    }
```

Comment:
could you move these 3 tests to a new file, `SKPaymentTransaction+ExtensionsTests`? Since they're not related to the `Purchases` class, and the test file is already pretty big

Reply:
yep, makes more sense

Reply:
Done ✅